packages/wallets/infinitywallet/src/adapter.ts

```diff
@@ -0,0 +1,242 @@
+import type { EventEmitter, SendTransactionOptions, WalletName } from '@solana/wallet-adapter-base';
+import {
+    BaseSignerWalletAdapter,
+    scopePollingDetectionStrategy,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSendTransactionError,
+    WalletSignMessageError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import type { Connection, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+
+interface InfinityWalletWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+interface InfinityWallet extends EventEmitter<InfinityWalletWalletEvents> {
+    isInfinityWallet?: boolean;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    getAccount(): Promise<string>;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signAndSendTransaction(
+        transaction: Transaction,
+        options?: SendOptions
+    ): Promise<{ signature: TransactionSignature }>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+}
+
+interface InfinityWalletWindow extends Window {
+    solana?: InfinityWallet;
```

Comment:
Please use a different namespace. New adapters that use `window.solana` aren't being accepted, a policy that has been applied for the entire lifetime of Wallet Adapter.

Reply:
This was marked as resolved but hasn't changed.

Reply:
It has been changed if you would like to look at the changed version and not the outdated version. It was changed to

```
InfinityWalletWindow extends Window {
    infinitywallet?: {
          solana?: InfinityWallet;
    };
    solana?: InfinityWallet;
}
```

Which we followed the exact same as Phantom since you mentioned all wallet should be equal.